### ISSUE TITLE
Massive performance decrease on BLOB column at data page

### ISSUE THREAD

**Issue by psharper** [suurjaak/SQLitely#4]
Hi,

using SQLitely version 2.3, 10.07.2024 on default Kubuntu 24.04 as a Snap.

When there is a BLOB column in the table filled with small binary data ~80KiB per row ~15 rows and opening a new Data tab page, the app hangs about 25s. Interaction with the tab page is also nearly impossible.

This did not happen, if you create the tab page by executing a sql query, because there the BLOB col is cut off.

**Comment by suurjaak** [comment]
Confirmed: data grids in Linux start lagging when columns have very long values. 

Will figure out a workaround for the problem. 

Thank you for reporting this.

**Comment by psharper** [comment]
Hi, 

another and imo corresponding issue "around" the wxpython gui frontend.

Executing the same sqlscript on the SQL-page multiple times. Everytime it gets slower! More complex the script more faster the slowdown. Until the popup "Running SQL ..." never disappears. But changing the focus to another tabpage and back immediately show the result of the query, so the backend still performs.

**Comment by suurjaak** [comment]
Is this still related to very long values in query results, or regardless of those?

**Comment by psharper** [comment]
No, it depends on VISIBLE rows at least O(n²) and slightly on visible cols in the result grid.
For my system 10 rows and 10 iterations are enough -> 1 core 100%.

Closing the grid at stress i occasionally get unexpected error "ToolBar has been deleted".

**Comment by suurjaak** [comment]
Sorry, I did not quite understand. Are you saying that if a query returns 10 result rows, it starts lagging?

Could you maybe share the database and the query?

**Comment by psharper** [comment]
The lagging depends on the VISIBLE cells of the wxGrid control and how often you hit the "Execute Query" button in the SAME "SQL" window openend by the SQL button on the left. The content of the cells is irrelevant.

After > 50 clicks lagging (on my system)
![pic1](https://github.com/user-attachments/assets/6e49b8c9-e10c-4aa1-aa78-1e54e2158025)

After 17 clicks lagging (on my system)
![pic2](https://github.com/user-attachments/assets/b841a38c-1601-4afa-9107-ac6ba56b8601)

After 3 clicks lagging (on my system)
![pic3](https://github.com/user-attachments/assets/a5fe581b-67c6-408b-852b-93d21d0e01d8)

In real world, using the program with normal queries, i have to change FOCUS to another SQL-Window to get the result and/or to make a new window for the next few query executions to not lag.

[test.zip](https://github.com/user-attachments/files/18040229/test.zip)

**Comment by suurjaak** [comment]
Thank you very much, could reproduce this lag myself now, and found the problem. 

I will try to get a new version released soon-ish; have been preparing for a release for some time already, but something always pops up.

**Comment by suurjaak** [comment]
The newly released version 2.4 has fixed these problems. 

Thank you for reporting them.
